test/test_md5.cpp

```diff
@@ -37,6 +49,9 @@ int main(int, char**)
         hash.process_bytes(expectations[i].data, expectations[i].len);
         boost::uuids::detail::md5::digest_type result;
         hash.get_digest(result);
+        for (boost::uint32_t j = 0; j < 4; ++j) {
```

Comment:
The byte-swapping is incorrect here (for any endian platform). 
There should be a direct memory comparison (memcmp, below) matching the expectations above.

Reply:
I still believe this comment is correct.  The result of get_digest should be the same in all cases, not conditional based on the platform.  The result is a series of raw bytes and therefore endian-unaware at this level.